src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -338,600 +347,343 @@ void InferRequestBase::changeDefaultPtr() {
-                                                    << ", but expect blobs with different shapes. Input shape: "
-                                                    << ov::PartialShape(blobDims) << ", output shape: " << shape;
+                        IE_THROW(ParameterMismatch)
+                            << "Network input and output use the same name: " << name
+                            << ", but expect tensors with different shapes. Input shape: " << ov::PartialShape(blobDims)
+                            << ", output shape: " << port_shape;
                     }
 
-                    const auto netOutPrc = InferenceEngine::details::convertPrecision(outputNode->second->get_input_element_type(0));
-                    if (netOutPrc != data->getTensorDesc().getPrecision()) {
+                    const auto netOutPrc = port.get_element_type();
+                    if (netOutPrc != tensor->get_element_type()) {
                         IE_THROW(ParameterMismatch)
-                                    << "Network input and output use the same name: " << name << " but expect blobs with different precision: "
-                                    << data->getTensorDesc().getPrecision() << " for input and " << netOutPrc
-                                    << " for output.";
+                            << "Network input and output use the same name: " << name
+                            << " but expect blobs with different precision: " << tensor->get_element_type()
+                            << " for input and " << netOutPrc << " for output.";
                     }
                 }
-
-                _outputs[name] = data;
-                if (!isDynamic && !externalPtr.count(name) &&
-                    data->getTensorDesc() == MemoryDescUtils::convertToTensorDesc(output->second->getParentEdgesAtPort(0)[0]->getMemory().getDesc())) {
-                    externalPtr[name] = data;
+                m_outputs[name] = tensor;
+                auto desc = create_tensor_desc(tensor);
+                if (!port_shape.is_dynamic() && !external_ptr.count(name) &&
+                    desc == MemoryDescUtils::convertToTensorDesc(
+                            output->second->getParentEdgesAtPort(0)[0]->getMemory().getDesc())) {
+                    external_ptr[name] = tensor;
                 }
             } else {
-                IE_THROW() << "Blob with name: " << name << " exists in CPU plugin graph, but absents in network outputs";
+                OPENVINO_THROW("Tensor with name: ", name, " exists in CPU plugin graph, but absents in network outputs");
+            }
+            // update tensors in case of multiple output ports with the same name
+            for (const auto& out : get_outputs()) {
+                auto port_name = get_port_name(out, m_is_legacy_api);
+                if ((name == port_name) && tensor && port != out) {
+                    ov::ISyncInferRequest::set_tensor(out, tensor);
+                }
```

Comment:
Plugin API 2.0 provides mechanism to make tensors can be shared, while tensor is bind to a tensor name, If multiple output port shared the same tensor, their name should be same.

Reply:
In ov::ISyncInferRequest, there is a map to manage port and its tensor:

port->port_id->descriptor::Tensor->ov::SoPtr<ov::ITensor>
https://github.com/maxnick/openvino/blob/7568b17d30f87b949dddb9d40156bc7d2c79535f/src/inference/src/dev/isync_infer_request.cpp#L177-L207

In case of different ports have different port_id with same port name, it means they shared the same tensor, we need set the tensor to this port, else  `m_tensors.at(get_inputs().at(found_port.idx).get_tensor_ptr())` will be empty and `ov::ISyncInferRequest::get_tensor` will return empty tensor.

Reply:
>>Also what about the first part of the comment (early exit)?

Yes, have refactored 2 weeks ago.
This function only is called once for each port, the flow for input and output:
It will check the port name whether in `graph->outputNodesMap` or `graph->inputNodesMap`, if it in any one, it will go its branch. Once into input or output branch, it will check the name whether in network port's tensor map, if cannot find then we need create tensor for this port.